### PR TITLE
[IA-5028] Increase default docker shared memory size

### PR DESCRIPTION
**Pull request by LizBaldo**
__Jira ticket__: https://broadworkbench.atlassian.net/browse/IA-5028

### What

- Increases the amount of shared memory allocated to the jupyter docker container. Bumping the shm size to 256m worked great, but we are attempting to dynamically adjust this value to about 50% of the total amount of memory allocated to the container

### Why

- The current default is 64mb, which is not sufficient for some users. Doing a dynamic allocation in the backend directly will avoid a lot of unnecessary UI and UX work, as exposing this might lead to a bad user experience (aka jupyter crashing)
### What to test

Tested on my BEE for both a GCE (jupyter and RStudio) runtime with and without GPU, as well as for a dataproc cluster, and can see the amount of memory matching what is requested in the docker composes:
Screenshot 2024-07-29 at 5 01 58 PM

### Who tested and where

- [ ] This change is covered by automated tests 
  - _NB: Rerun automation tests on this PR by commenting `jenkins retest` or `jenkins multi-test`._
- [ ] I validated this change 
- [ ] Primary reviewer validated this change 
- [ ] I validated this change in the dev environment

### CODE REVIEW

File: http/src/main/resources/init-resources/startup.sh
Comment (LizBaldo) [inline http/src/main/resources/init-resources/startup.sh:131]: I don't know why, but by default the docker compose file was not being pulled when restarting the runtime. That can be problematic when users are trying to update their machine with different memory and CPU allocations.

File: http/src/main/resources/init-resources/gce-init.sh
Comment (LizBaldo) [inline http/src/main/resources/init-resources/gce-init.sh]: I know I could do it within the scala code in leo, but the list of runtime template values is already sooo long that it felt cleaner this way honestly, but happy to rework if you think otherwise, I just want to get this in front of users ASAP to help unblock them

Comment (lucymcnatt) [reply http/src/main/resources/init-resources/gce-init.sh]: totally fine with this approach 👍

**Review by lucymcnatt — APPROVED** [review]
Looks great!

File: http/src/main/resources/init-resources/gce-init.sh
Comment (jdcanas) [reply http/src/main/resources/init-resources/gce-init.sh]: hmm... I would honestly prefer to do it in leo. I find whenever I use `awk` it feels elegant to start with, but a year or so down the line its broken by bash's lack of type safety in conjunction with an assumption being changed somewhere else in the application (i.e. `memLimit` changing data types in this case).

Comment (jdcanas) [reply http/src/main/resources/init-resources/gce-init.sh]: also, our init scripts aren't tested very well 😬

**Review by jdcanas — APPROVED** [review]
would like to avoid logic in the init script like this since it isn't tested, but not going to hold the PR on it

File: http/src/main/resources/init-resources/gce-init.sh
Comment (LizBaldo) [reply http/src/main/resources/init-resources/gce-init.sh]: This is a very good point, I'll take a stab at moving the logic out of the init scripts and into Leonardo proper with unit testing tomorrow.

**Comment by LizBaldo** [comment]
@jdcanas I made the changes and moved the logic so it an be unit testable. However I can't merge it because the integration tests can't run due to a path being deleted `ls: cannot access '/home/runner/work/terra-github-workflows/terra-github-workflows/leonardo/test-reports': No such file or directory`

I am not going to be around next week, neither will Lucy, so would you be able to:
- Rerun the tests on Monday and see if the underlying infra issue is still there?
- Ask Devops about it if it is still a problem
- Merge the PR and keep an eye on things if it needs to be reverted?


Many thanks 🙏

File: http/src/main/resources/init-resources/gce-init.sh
Comment (LizBaldo) [reply http/src/main/resources/init-resources/gce-init.sh]: @jdcanas I took a stab at it if you want to re-review and merge while I am away, thanks!

**Pull request merged**
